### ISSUE TITLE
Questions about cross-entropy loss

### ISSUE THREAD

**Issue by weiyithu** [cv-rits/MonoScene#26]
Dear authors, thanks for your great works! In your paper, you say that "the losses are computed only where y is
defined". I wonder if this means you do not add supervision on non-occupied voxels and only use multi-class classification loss on occupied voxels ? If this holds true, why the model can identify which voxels are occupied ?

**Comment by anhquancao** [comment]
Hi @weiyithu, "y is defined" means that we know the class of the voxels which could be either non-occupied or occupied. There is another type of voxels called "unknown voxels" that we don't know its class since they were not observed in any lidar frames used to accumulate the ground-truth.

**Comment by weiyithu** [comment]
Thanks, I get it ! If I only use cross-entroy loss without using other losses introduced your paper, can the model converge？ And how do you calculate the loss weight of each class, such as non-occupied class?

**Comment by anhquancao** [comment]
Yes, the model can converge but the performance is around 2 mIoU lower. The weight of each class is just 1/class_frequencies(number of voxels per class in the whole dataset)
https://github.com/cv-rits/MonoScene/blob/master/monoscene/data/semantic_kitti/params.py#L3

**Comment by weiyithu** [comment]
Thanks! Last question, have you tried only use occupied / non-occupied as the supervision instead of multi-class classification? I wonder if the semantic supervsion is necessary for the occupancy prediction?

**Comment by anhquancao** [comment]
Yes, I tried, the occupancy performance is roughly similar.

**Comment by weiyithu** [comment]
You mean that if you use occupied / non-occupied (two class) supervision signal, you can still a model which has the good performance on occupancy prediction?  In my experiment, I find that if I only use occupied / non-occupied  supervsion, the model can only predict good ground, since the most of points are ground points and there exists data imbalanced problem. So I wonder if the semantic  labels can solve this problem.

**Comment by anhquancao** [comment]
I never tried with only cross entropy loss. I tried only with full losses

**Comment by weiyithu** [comment]
My task is not SSC but is similar to SSC and my model does not converge, so I have not evaluate IoU. The full losses mean that you use both cross-entropy loss and the loss introduced in your paper? And the full losses have semantic information?

**Comment by anhquancao** [comment]
I use the full loss with only 2 classes: 0 for non-occupied and 1 for occupied. Actually, I haven't draw the results but I got IoU of around 30%.

**Comment by weiyithu** [comment]
OK. Thanks for your patience！Great Work！！

**Comment by weiyithu** [comment]
OK. Thanks for your patience！Great Work！！

**Comment by weiyithu** [comment]
OK. Thanks for your patience！Great Work！！

**Comment by anhquancao** [comment]
You're welcome!
